libpod/reset.go

```diff
@@ -0,0 +1,99 @@
+package libpod
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/containers/libpod/libpod/define"
+	"github.com/containers/libpod/pkg/rootless"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Reset removes all storage
+func (r *Runtime) Reset() error {
+	ctx := context.TODO()
+
+	pods, err := r.GetAllPods()
+	if err != nil {
+		return err
+	}
+	for _, p := range pods {
+		if err := r.RemovePod(ctx, p, true, true); err != nil {
+			if errors.Cause(err) == define.ErrNoSuchPod {
+				continue
+			}
+			logrus.Errorf("Error removing Pod %s: %v", p.ID(), err)
+		}
+	}
+
+	ctrs, err := r.GetAllContainers()
+	if err != nil {
+		return err
+	}
+
+	for _, c := range ctrs {
+		if err := r.RemoveContainer(ctx, c, true, true); err != nil {
+			if err := r.RemoveStorageContainer(c.ID(), true); err != nil {
+				if errors.Cause(err) == define.ErrNoSuchCtr {
+					continue
+				}
+				logrus.Errorf("Error removing container %s: %v", c.ID(), err)
+			}
+		}
+	}
+
+	ir := r.ImageRuntime()
+	images, err := ir.GetImages()
+	if err != nil {
+		return err
+	}
+
+	for _, i := range images {
```

Comment:
i cannot remember but is it possible that rm -fa on images would remove all containers and pods?

Reply:
Um. It would try, but it wouldn't be able to remove infra containers, so everything would be gone except infra containers and pods.